x-pack/plugins/transform/common/api_schemas/common.ts

```diff
@@ -58,17 +56,27 @@ export interface CommonResponseStatusSchema {
 }
 
 export const runtimeMappingsSchema = schema.maybe(
-  schema.object(
-    {},
-    {
-      unknowns: 'allow',
-      validate: (v: object) => {
-        if (Object.values(v).some((o) => !isRuntimeField(o))) {
-          return i18n.translate('xpack.transform.invalidRuntimeFieldMessage', {
-            defaultMessage: 'Invalid runtime field',
-          });
-        }
-      },
-    }
+  schema.recordOf(
```

Comment:
I don't think this change is needed isRuntimeField() is already be supporting 'composite' if it's listed in data.common's `RUNTIME_FIELD_TYPES`. Since users can create transform through API, we have to support more cases than what Kibana currently supports.